src/entity/units/goru.ts

```diff
@@ -44,50 +60,328 @@ const unit: UnitSource = {
     death: 'goruDeath'
   },
   init: (unit: Unit.IUnit, underworld: Underworld) => {
+    // TODO - Bug: Undying 2 can be re-added by cloning/splitting/creating a Goru with no undying modifier
+    // Using originalLife might prevent this, but also prevents summoned Goru's from getting undying.
+    if (!unit.modifiers[undyingModifierId]) {
+      Unit.addModifier(unit, undyingModifierId, underworld, false, 2);
+    }
   },
   action: async (unit: Unit.IUnit, attackTargets: Unit.IUnit[] | undefined, underworld: Underworld, canAttackTarget: boolean) => {
-    const attackTarget = attackTargets && attackTargets[0];
-    // Attack
-    if (attackTarget && canAttackTarget) {
-      unit.mana -= unit.manaCostToCast;
-      Unit.orient(unit, attackTarget);
-      await Unit.playComboAnimation(unit, unit.animations.attack, () => {
-        return createVisualLobbingProjectile(
-          unit,
-          attackTarget,
-          'projectile/lobberProjectile',
-        ).then(() => {
-          if (attackTarget) {
-            Unit.takeDamage({
-              unit: attackTarget,
-              amount: unit.damage,
-              sourceUnit: unit,
-              fromVec2: unit,
-            }, underworld, false);
-            // Add projectile hit animation
-            Image.addOneOffAnimation(attackTarget, 'projectile/lobberProjectileHit');
+    // Goru has 2 "action points" and 5 actions to choose from
+    // Actions are prioritized as follows:
+    // - 2AP - Consume/Explode/Resurrect primed corpses
+    // - 2AP - Prime corpses (minimum 3)
+    // - 1AP - Attack enemies
+    // - 1AP - Summon allies
+    // - 1AP - Move
+    let actionPoints = 2;
+
+    // - Action -
+    // Consume/Explode/Resurrect primed corpses
+    if (actionPoints >= 2 && unit.mana >= unit.manaCostToCast) {
+      const primedCorpses = underworld.units.filter(u => u.modifiers[primedCorpseId])
```

Comment:
Let's make sure to QA primed corpses with save/load and on multiplayer if you haven't already